specs/eventing/control-plane.md

```diff
@@ -0,0 +1,1007 @@
+  </tr>
+</table>
+
+### Addressable
+
+Note that the Addressable interface is a partial schema -- any resource which
+includes these fields MAY be referenced using a `duckv1.Destination`.
+
+#### Metadata:
+
+Standard Kubernetes
+[metav1.ObjectMeta](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.21/#objectmeta-v1-meta)
+resource. Note that there are no restrictions on `apiVersion` or `kind` as long
+as the object matches the partial schema.
+
+#### Spec:
+
+There are no `spec` requirements for Addressable.
+
+#### Status
+
+<table>
+  <tr>
+    <td><strong>Field Name</strong></td>
+    <td><strong>Field Type</strong></td>
+    <td><strong>Description</strong></td>
+    <td><strong>Schema Requirement</strong></td>
+  </tr>
+  <tr>
+    <td><code>address</code></td>
+    <td><a href="#duckv1addressable">duckv1.Addressable</a></td>
+    <td>Address used to deliver events to the resource.</td>
+    <td>REQUIRED</td>
+  </tr>
+</table>
+
+## Detailed SubResource Objects
+
+### duckv1.Addressable
+
+<table>
+  <tr>
+    <td><strong>Field Name</strong></td>
+    <td><strong>Field Type</strong></td>
+    <td><strong>Description</strong></td>
+    <td><strong>Schema Requirement</strong></td>
+  </tr>
+  <tr>
+    <td><code>url</code></td>
+    <td>URL (string)</td>
+    <td>Address used to deliver events to the Addressable.</td>
+    <td>REQUIRED</td>
+  </tr>
+</table>
+
+### duckv1.Destination
+
+Destination is used to indicate the destination for event delivery. A
+Destination eventually resolves the supplied information to a URL by resolving
+`uri` relative to the address of `ref` (if provided) as described in
+[Addressable resolution](#addressable-resolution). `ref` MAY be an
+[Addressable](#duckv1-addressable) object or a `v1/Service` Kubernetes service.
+
+<table>
+  <tr>
+    <td><strong>Field Name</strong></td>
+    <td><strong>Field Type</strong></td>
+    <td><strong>Description</strong></td>
+    <td><strong>Schema Requirement</strong></td>
+  </tr>
+  <tr>
+    <td><code>ref</code></td>
+    <td><a href="#duckv1kreference">duckv1.KReference</a><br/>(OPTIONAL)</td>
+    <td>An ObjectReference to an Addressable reference to deliver events to.</td>
+    <td>REQUIRED</td>
+  </tr>
+  <tr>
+    <td><code>uri</code></td>
+    <td>URL (string)<br/>(OPTIONAL)</td>
+    <td>A resolved URL to deliver events to.</td>
+    <td>REQUIRED</td>
+  </tr>
+</table>
+
+### duckv1.SubscriberSpec
+
+SubscriberSpec represents an automatically-populated extraction of information
+from a [Subscription](#subscription).
+
+<table>
+  <tr>
+    <td><strong>Field Name</strong></td>
+    <td><strong>Field Type</strong></td>
+    <td><strong>Description</strong></td>
+    <td><strong>Schema Requirement</strong></td>
+  </tr>
+  <tr>
+    <td><code>uid</code></td>
+    <td>UID (string)</td>
+    <td>UID is used to disambiguate Subscriptions which might be recreated.</td>
+    <td>REQUIRED</td>
+  </tr>
+  <tr>
+    <td><code>generation</code></td>
```

Comment:
Like "observedGeneration" I'd prefer to remove this, but if we keep both then I think they need to be required fields